File: backend/app/services/themes.py
from pydantic import BaseModel


class Theme(BaseModel):
    id: str
    name: str
    description: str
    accent: str
    background: str
    font_heading: str
    font_body: str
    cover_emoji: str


THEMES: list[Theme] = [
    Theme(
        id="pitch-classic",
        name="Pitch — Classic",
        description="Investor-friendly default. Crisp serif headings, plenty of whitespace.",
        accent="#EC4899",
        background="#FFFFFF",
        font_heading="\"Source Serif Pro\", serif",
        font_body="Inter, system-ui, sans-serif",
        cover_emoji="🚀",
    ),
    Theme(
        id="pitch-bold",
        name="Pitch — Bold",
        description="High-contrast, oversized headlines. For Series A founders.",
        accent="#FACC15",
        background="#0F172A",
        font_heading="\"Space Grotesk\", sans-serif",
        font_body="Inter, system-ui, sans-serif",
        cover_emoji="⚡",
    ),
    Theme(
        id="report-minimal",
        name="Report — Minimal",
        description="Quiet, data-dense layout for quarterly reports.",
        accent="#0EA5E9",
        background="#F8FAFC",
        font_heading="Inter, system-ui, sans-serif",
        font_body="Inter, system-ui, sans-serif",
        cover_emoji="📊",
    ),
    Theme(
        id="training-warm",
        name="Training — Warm",
        description="Friendly tone, large body text, ideal for onboarding decks.",
        accent="#F97316",
        background="#FFF7ED",
        font_heading="\"Plus Jakarta Sans\", sans-serif",
        font_body="\"Plus Jakarta Sans\", sans-serif",
        cover_emoji="🎓",
    ),
    Theme(
        id="dark-investor",
        name="Dark — Investor",
        description="Late-night demo aesthetic. White text on near-black.",
        accent="#A78BFA",
        background="#0B0B12",
        font_heading="\"IBM Plex Sans\", sans-serif",
        font_body="\"IBM Plex Sans\", sans-serif",
        cover_emoji="🌙",
    ),
]


def list_themes() -> list[Theme]:
    return THEMES


def get_theme(theme_id: str) -> Theme | None:
    return next((t for t in THEMES if t.id == theme_id), None)

File: web/src/lib/deck/themes.ts
import type { DeckJson } from "./types";

export interface ResolvedTheme {
  id: string;
  name: string;
  description: string;
  accent: string;
  bg: string;
  surface: string;
  text: string;
  muted: string;
  border: string;
  fontBody: string;
  fontHeading: string;
  isDark: boolean;
  emoji: string;
}

const SERIF = '"Source Serif 4", Georgia, serif';
const SANS = 'Inter, system-ui, -apple-system, sans-serif';

export const THEMES: ResolvedTheme[] = [
  {
    id: "slide-dark",
    name: "Slide — Dark",
    description: "The DClaw default. White on near-black, pink accent.",
    accent: "#EC4899",
    bg: "#18181b",
    surface: "rgba(39,39,42,0.6)",
    text: "#f4f4f5",
    muted: "#a1a1aa",
    border: "#27272a",
    fontBody: SANS,
    fontHeading: SANS,
    isDark: true,
    emoji: "💗",
  },
  {
    id: "pitch-classic",
    name: "Pitch — Classic",
    description: "Investor-friendly. Serif headings, lots of whitespace.",
    accent: "#EC4899",
    bg: "#ffffff",
    surface: "#f4f4f5",
    text: "#18181b",
    muted: "#52525b",
    border: "#e4e4e7",
    fontBody: SANS,
    fontHeading: SERIF,
    isDark: false,
    emoji: "🚀",
  },
  {
    id: "pitch-bold",
    name: "Pitch — Bold",
    description: "High-contrast, oversized headlines for Series A.",
    accent: "#FACC15",
    bg: "#0f172a",
    surface: "rgba(30,41,59,0.7)",
    text: "#f8fafc",
    muted: "#94a3b8",
    border: "#1e293b",
    fontBody: SANS,
    fontHeading: SANS,
    isDark: true,
    emoji: "⚡",
  },
  {
    id: "report-minimal",
    name: "Report — Minimal",
    description: "Quiet, data-dense layout for quarterly reports.",
    accent: "#0EA5E9",
    bg: "#f8fafc",
    surface: "#eef2f6",
    text: "#0f172a",
    muted: "#475569",
    border: "#e2e8f0",
    fontBody: SANS,
    fontHeading: SANS,
    isDark: false,
    emoji: "📊",
  },
  {
    id: "training-warm",
    name: "Training — Warm",
    description: "Friendly tone, large body text for onboarding.",
    accent: "#F97316",
    bg: "#fff7ed",
    surface: "#ffedd5",
    text: "#431407",
    muted: "#9a3412",
    border: "#fed7aa",
    fontBody: SANS,
    fontHeading: SANS,
    isDark: false,
    emoji: "🎓",
  },
  {
    id: "dark-investor",
    name: "Dark — Investor",
    description: "Late-night demo aesthetic. Violet on black.",
    accent: "#A78BFA",
    bg: "#0b0b12",
    surface: "rgba(30,27,46,0.7)",
    text: "#ede9fe",
    muted: "#a5a3c0",
    border: "#241f33",
    fontBody: SANS,
    fontHeading: SANS,
    isDark: true,
    emoji: "🌙",
  },
];

export const DEFAULT_THEME = THEMES[0];

/** Resolve a deck's stored theme to concrete styling. */
export function resolveTheme(theme: DeckJson["theme"] | undefined): ResolvedTheme {
  if (theme?.id) {
    const found = THEMES.find((t) => t.id === theme.id);
    if (found) return found;
  }
  // Back-compat for decks stored before named themes: derive from accent/bg.
  if (theme) {
    const dark = theme.background !== "light";
    const base = dark ? THEMES[0] : THEMES[1];
    return {
      ...base,
      id: "custom",
      name: "Custom",
      accent: theme.accent ?? base.accent,
      fontHeading: theme.font === "serif" ? SERIF : base.fontHeading,
    };
  }
  return DEFAULT_THEME;
}
